### PR TITLE
Document R.ap as S Combinator

### PR DESCRIPTION
**Pull request by MikaAK**
#1916 Redo since I lost access to original repo. Not sure if you want signatures just let me know which one of these to use

```
ap :: Function x (a -> b) -> Function x a -> Function x b
```
```
ap :: (x -> a -> b) -> (x -> a) -> (x -> b)
```
```
ap :: (a -> b -> c) -> (a -> b) -> (a -> c)
```

### CODE REVIEW

File: src/ap.js [hunk 1]
```diff
@@ -23,6 +23,10 @@ var map = require('./map');
  *
  *      R.ap([R.multiply(2), R.add(3)], [1,2,3]); //=> [2, 4, 6, 4, 5, 6]
  *      R.ap([R.concat('tasty '), R.toUpper], ['pizza', 'salad']); //=> ["tasty pizza", "tasty salad", "PIZZA", "SALAD"]
+ *
+ *      // R.ap can also be used as S combinator
+ *      // when only two functions are passed
+ *      R.ap(R.concat, R.toUpper)('Ramda') //=> 'RamdaRAMDA'
```
Comment (davidchambers) [inline src/ap.js]: I'd prefer `, ` in place of `)(`.

**Comment by CrossEye** [comment]
I think I'd prefer this one:

```hs
ap :: (a -> b -> c) -> (a -> b) -> (a -> c)
```

File: src/ap.js (hunk 1 above)
Comment (kedashoe) [inline src/ap.js]: Like `R.ap(R.concat, R.toUpper, 'Ramda')`? I don't think we can call it like that

Comment (davidchambers) [inline src/ap.js]: You're quite right, Kevin! Ignore my comment, @MikaAK. :)

**Comment by kedashoe** [comment]
@MikaAK would you like to add the signature specialized for functions? Please squash as well :)

**Comment by MikaAK** [comment]
@kedashoe  that's `ap :: (a -> b -> c) -> (a -> b) -> (a -> c)`? sorry I'm not very good with type sigs yet :(

**Comment by kedashoe** [comment]
> that's ap :: (a -> b -> c) -> (a -> b) -> (a -> c)

yep :)

**Comment by MikaAK** [comment]
Hopefully I got that one right! You should be able to squash on merge using the github ui 😄  but I can totally do it manually too if required!

**Pull request merged**